CMakeLists.txt

```diff
@@ -1007,6 +1021,13 @@ set(CPACK_WIX_PROPERTY_ARPHELPLINK "${CPACK_PACKAGE_HOMEPAGE_URL}")
 set(CPACK_WIX_UI_BANNER "${CMAKE_CURRENT_SOURCE_DIR}/build/wix/images/banner.bmp")
 set(CPACK_WIX_UI_DIALOG "${CMAKE_CURRENT_SOURCE_DIR}/build/wix/images/dialog.bmp")
 
+if(UNIX AND NOT APPLE)
+  add_subdirectory(build/debian)
+  add_subdirectory(build/rpm)
+endif()
```

Comment:
The code from these `CMakeLists.txt` files should go into a `CPackConfigDEB.cmake` and `CPackConfigRPM.cmake` inside the `cmake` directory instead.

Then, you can include them from the `CPackConfig.cmake` file (not `CMakeLists.txt`) like this:

    if(CPACK_GENERATOR MATCHES "DEB")
        include(CPackConfigDEB)
    endif()

Reply:
I am not sure if this works, because CPackConfig.cmake is used at packaging time. Did you try it? 

But anyway, the current structure is done in a way that the project specific settings are in CMakeList.txt files in context at the debian and rpm folder while the common files that can be used in any project live in the cmake folder. I would like to keep this. 

What is the bad thing with add_subdirectory here?